realm/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -0,0 +1,383 @@
+         * Sets the {@link io.realm.RealmMigration} to be run if a migration is needed. If this migration fails to
+         * upgrade the on-disc schema to the runtime schema, a
+         * {@link io.realm.exceptions.RealmMigrationNeededException} will be thrown.
+         */
+        public Builder migration(RealmMigration migration) {
+            if (migration == null) {
+                throw new IllegalArgumentException("A non-null migration must be provided");
+            }
+            this.migration = migration;
+            return this;
+        }
+
+        /**
+         * Setting this will change the behavior of how migration exceptions are handled. Instead of throwing a
+         * {@link io.realm.exceptions.RealmMigrationNeededException} the on-disc Realm will be cleared and recreated
+         * with the new Realm schema.
+         *
+         * <bold>WARNING!</bold> This will result in loss of data.
+         */
+        public Builder deleteRealmIfMigrationNeeded() {
+            this.deleteRealmIfMigrationNeeded = true;
+            return this;
+        }
+
+        /**
+         * Add a {@link RealmModule}s to the existing modules. RealmClasses in the new module is added to the schema
+         * for this Realm.
+         *
+         * @param module {@link RealmModule} to add to this Realms schema.
+         *
+         * @throws {@link IllegalArgumentException} if module is {@code null} or doesn't have the {@link RealmModule}
+         * annotation.
+         */
+        public Builder addModule(Object module) {
+            checkModule(module);
+            modules.add(module);
+            return this;
+        }
+
+        /**
+         * Replace the existing module(s) with one or more {@link RealmModule}s. Using this method will replace the
+         * current schema for this Realm with the schema defined by the provided modules.
+         *
+         * @param baseModule
+         * @param additionalModules
+         *
+         * @throws {@link IllegalArgumentException} if any of the modules are {@code null} or doesn't have the
+         * {@link RealmModule} annotation.
+         */
+        public Builder setModules(Object baseModule, Object... additionalModules) {
+            modules.clear();
+            addModule(baseModule);
+            if (additionalModules != null) {
+                for (int i = 0; i < additionalModules.length; i++) {
+                    Object module = additionalModules[i];
+                    checkModule(module);
+                    addModule(module);
+                }
+            }
+            return this;
+        }
+
+        /**
+         * DEBUG method. This restricts the Realm schema to only consist of the provided classes without having to
+         * create a module. These classes must be available in the default module. Calling this will remove any
+         * previously configured modules.
+         */
+        Builder schema(Class<? extends RealmObject>... schemaClass) {
```

Comment:
Even if it's just a method for debugging purposes, we should still use varargs properly